### ISSUE TITLE
[Accessibility Audit] Code Intel: Editing auto-indexing policies for a repository

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#33516]
### Steps to replicate journey

Taken from the [user journey list document](https://docs.google.com/document/d/1kA6aVOAgID_uPm-d6uEC1DG6WHjJ8fPT0UAXQcML_KQ/edit?usp=sharing). Use this for further context.

- Navigate to a repository in Sourcegraph
- Click on the 🧠 Code Intelligence button.
- Click on Configuration policies.
- Potential options:
  - Click on an existing policy.
  - Click on Create New Policy.
    - Fill out various fields in form and click Create Policy.

### How to audit

Follow the instructions here: [Auditing a user journey](https://docs.sourcegraph.com/dev/background-information/web/accessibility/how-to-audit#auditing-a-user-journey)

**Note:** We are planning on arranging thorough third-party audit in the future, so our focus here is just to find the _majority_ of accessibility issues. Don't worry if you feel you haven't found 100% of potential issues, it is better to focus on the core essentials to complete the journey rather than spending lots of time going through every possible problem.

### How to raise discovered problems

Follow the instructions here: [Raising an accessibility bug](https://docs.sourcegraph.com/dev/background-information/web/accessibility/how-to-audit#raising-a-bug)

### Any other questions?

See the [audit FAQ](https://docs.google.com/document/d/1nCi7TvYRpj9eG0_QtgqK1c4L28EiJKVuxyuvg1lAGgo/edit#) - add your question if it isn't there already!

**Comment by sourcegraph-bot-2** [comment]
Heads up @macraig - the "team/code-intelligence" label was applied to this issue.

**Comment by Numbers88s** [comment]
**Keyboard navigation**

_Main page_
It was difficult to know whether you are in the navigation list. 

Proposed Changes:
- It might be worth it to add more labels. 

_Precise code intelligence configuration page_

Post navigation it's difficult to know where you are, without restarting from the top. It's difficult to know what your focus is currently or where you are.

Need to fix: 
- aria-label is wrong it says "sort notebooks" for the Filtered component

Proposed Changes:
 - The link should say a description instead of the hash
 - disabled text inputs are described as “dimmed”. Don’t know if this is screen reader convention?
 - when there are is concatenating of string it reads it as different component and it breaks speech delivery.
 -  Implement best practices around labels and required fields. 
 - Remove the use of stepper (input with type of number) as it doesn't call out the updated amount, just the change in number. Quite confusing and misleading. 
 - It should announce the new header where you navigated to after clicking on a link.  I want to know what I opened.
 - Add aria-label for N/A of “Not applicable“
- Call out the number of results based on text entered on input field
